sourcegraph/sourcegraph
# dotcom: don't lazy add repos with `size >= 1GB && stars < 100`

This adds a new restriction to the on-demand cloning mechanism that's used on sourcegraph.com: when a repository has `>= 1GB` in size and `stars < 100` and it's hosted on GitHub.com, then we don't add it and show an error message.

When a repository runs into this restriction, we show the following error message:

<img width="1220" alt="screenshot_2023-12-05_15 23 03@2x" src="https://github.com/sourcegraph/sourcegraph/assets/1185253/bafb0748-bb50-4295-8822-7cb269da8d48">


## Test plan

- Ensure your `external-services-config.json` in `dev-private` contains an external service pointing to GitHub with `cloudDefault: true`:

```json
{
  "GITHUB": [
    {
      "url": "https://github.com",
      "token": "<TOKEN>",
      "cloudDefault": true,
      "repos": [ ]
    }
  ]
}
```
- Start sourcegraph in dotcom-mode: `sg start dotcom`
- Visit https://sourcegraph.test:3443/github.com/depop/documentation -- should *NOT* be cloned, too big and too few stars
- Visit https://sourcegraph.test:3443/github.com/torvalds/linux -- this should clone, since it has enough stars
